### PR TITLE
Fixed empty locale when generating property names

### LINKED ISSUE #1704: Properties without locale in database
I have a lot of pages with properties (in phpcr) which have a name like this: `i18n:-excerpt-title` i think it only happens with extensions but i am not quite sure.

### PR DESCRIPTION
**Pull request by wachterjohannes**
**tasks:**
- [x] test coverage
- [ ] gather feedback for my changes

**informations:**

| q | a |
| --- | --- |
| Fixed tickets | fixes https://github.com/sulu-io/sulu/issues/1704 |
| BC breaks | none |
| Documentation PR | none |

### CODE REVIEW

File: lib/Subscriber/Core/MappingSubscriber.php [hunk 1]
```diff
@@ -111,14 +116,21 @@ private function persistReference(NodeInterface $node, DocumentAccessor $accesso
...
 
         $referenceNode = $this->documentRegistry->getNodeForDocument($referenceDocument);
-        $phpcrName = $this->encoder->encode($fieldMapping['encoding'], $fieldMapping['property'], $locale);
+        try {
+            $phpcrName = $this->encoder->encode($fieldMapping['encoding'], $fieldMapping['property'], $locale);
+        } catch (\InvalidArgumentException $ex) {
+            // arguments unvalid no valid propertyname could be generated (e.g. no locale given for localized encoding)
+            return;
+        }
         $node->setProperty($phpcrName, $referenceNode);
```
Comment (danrot) [inline lib/Subscriber/Core/MappingSubscriber.php]: Why don't you just move that also into the try block, and therefore remove the `return` statement in the catch block? I think we said we either return early or at the very end, that's somewhere in between.

File: lib/Subscriber/Core/MappingSubscriber.php [hunk 2]
```diff
@@ -131,9 +143,19 @@ private function persistReference(NodeInterface $node, DocumentAccessor $accesso
...
+        DocumentAccessor $accessor,
+        $fieldName,
+        $locale,
+        array $fieldMapping
+    ) {
+        try {
+            $phpcrName = $this->encoder->encode($fieldMapping['encoding'], $fieldMapping['property'], $locale);
+        } catch (\InvalidArgumentException $ex) {
+            // arguments unvalid no valid propertyname could be generated (e.g. no locale given for localized encoding)
+            return;
```
Comment (danrot) [inline lib/Subscriber/Core/MappingSubscriber.php:161]: Same as above

File: lib/Subscriber/Core/MappingSubscriber.php [hunk 3]
```diff
@@ -184,10 +206,22 @@ public function handleHydrate(AbstractMappingEvent $event)
...
```
Comment (danrot) [inline lib/Subscriber/Core/MappingSubscriber.php:233]: Same as above

File: lib/Subscriber/Core/MappingSubscriber.php [hunk 4]
```diff
@@ -208,9 +242,19 @@ private function hydrateReferenceField(NodeInterface $node, $document, DocumentA
...
```
Comment (danrot) [inline lib/Subscriber/Core/MappingSubscriber.php:262]: Same as above

**Comment by danrot** [comment]
Do we need the catch blocks at all? Shouldn't these exceptions never be thrown, if all the subscribers are working correctly?

**Comment by wachterjohannes** [comment]
this is an own subscriber which can work without locale (unlocalized properties). so it has to be checked here.

**Comment by wachterjohannes** [comment]
@danrot fixed

**Comment by wachterjohannes** [comment]
@danrot fixed

**Comment by danrot** [comment]
Shouldn't that PR be targeted at develop?

**Comment by wachterjohannes** [comment]
yes i will change this

**Pull request closed without merging**